<a href="https://colab.research.google.com/github/bobs24/Data-Science-Learning-Path/blob/Learning/DataFrame_Pandas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [ ]:
# Membuat dataframes dari dictionary
import pandas as pd
Trial = pd.DataFrame({'Uangmu':[26,52],'Uangku':[100,500]})
print(Trial)

   Uangmu  Uangku
0      26     100
1      52     500


<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [ ]:
# Membaca data dan memasukannya kedalam suatu variabel
df_from_csv=pd.read_csv('diamonds.csv')
print(df_from_csv)

       Unnamed: 0  carat        cut color  ... price     x     y     z
0               1   0.23      Ideal     E  ...   326  3.95  3.98  2.43
1               2   0.21    Premium     E  ...   326  3.89  3.84  2.31
2               3   0.23       Good     E  ...   327  4.05  4.07  2.31
3               4   0.29    Premium     I  ...   334  4.20  4.23  2.63
4               5   0.31       Good     J  ...   335  4.34  4.35  2.75
...           ...    ...        ...   ...  ...   ...   ...   ...   ...
53935       53936   0.72      Ideal     D  ...  2757  5.75  5.76  3.50
53936       53937   0.72       Good     D  ...  2757  5.69  5.75  3.61
53937       53938   0.70  Very Good     D  ...  2757  5.66  5.68  3.56
53938       53939   0.86    Premium     H  ...  2757  6.15  6.12  3.74
53939       53940   0.75      Ideal     D  ...  2757  5.83  5.87  3.64

[53940 rows x 11 columns]


Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

Jawab:  
Series-List = dapat dilakukan metode slicing dan mengakses object series
series-dictionary = index yang ada di series berfungsi seperti key di dictionary

dataframe terdiri dari 2 index (baris-kolom) jumlah index harus sama dengan jumlah data, antar kolom bisa punya data yang berbeda tapi 1 kolom hanya 1 jenis data.
series terdiri dari 1 index (string,int,bool,etc)

In [ ]:
# code here
import pandas as pd
x = pd.Series([1,2,3,4,5])
print(x)

0    1
1    2
2    3
3    4
4    5
dtype: int64


<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [ ]:
# code here
df_from_csv=pd.read_csv('diamonds.csv')
print(df_from_csv['price'][0])
print(df_from_csv['color'][53939])

326
D


<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

Jawab: metode pengumpulan dan visualisasi/penyajian data (menggunakan tabel,report, diagram, grafik), bisa berupa numerik.

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [ ]:
# code here
import pandas as pd
df=pd.read_csv('diamonds.csv')
print(df['price'].mean())
print(df['depth'].std())
print(df['carat'].max())

3932.799721913237
1.4326213188337733
5.01


Jawab:
rata-rata price = 3932.79
std dev depth = 1.43
nilai maksimal carat = 5.01